stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -964,6 +965,74 @@ Return the updated `y`.
 """
 sbmv!
 
+### spmv!, (SP) symmetric packed matrix-vector operation defined as y := alpha*A*x + beta*y.
+for (fname, elty) in ((:dspmv_, :Float64),
+                      (:sspmv_, :Float32))
+    @eval begin
+        function spmv!(uplo::AbstractChar,
+                       n::BlasInt,
+                       α::$elty,
+                       AP::Union{Ptr{$elty}, AbstractArray{$elty}},
+                       x::Union{Ptr{$elty}, AbstractArray{$elty}},
+                       incx::Integer,
+                       β::$elty,
+                       y::Union{Ptr{$elty}, AbstractArray{$elty}},
+                       incy::Integer)
+
+            ccall((@blasfunc($fname), libblas), Cvoid,
+                  (Ref{UInt8},     # uplo,
+                   Ref{BlasInt},   # n,
+                   Ref{$elty},     # α,
+                   Ptr{$elty},     # AP,
+                   Ptr{$elty},     # x,
+                   Ref{BlasInt},   # incx,
+                   Ref{$elty},     # β,
+                   Ptr{$elty},     # y, out
+                   Ref{BlasInt}),  # incy
+                  uplo,
+                  n,
+                  α,
+                  AP,
+                  x,
+                  incx,
+                  β,
+                  y,
+                  incy)
+        end
+    end
+end
+
+function spmv!(uplo::AbstractChar,
+               α::Number, AP::Union{DenseArray{T}, AbstractVector{T}}, x::Union{DenseArray{T}, AbstractVector{T}},
+               β::Number, y::Union{DenseArray{T}, AbstractVector{T}}) where {T <: BlasReal}
+    require_one_based_indexing(AP, x, y)
+    N = length(x)
+    if N != length(y)
+        throw(DimensionMismatch("x has length $(N), but y has length $(length(y))"))
+    end
+    if length(AP) < Int64(N*(N+1)/2)
+        throw(DimensionMismatch("Packed symmetric matrix A has size smaller than length(x) = $(N)."))
+    end
+    GC.@preserve x y AP spmv!(uplo, N, convert(T, α), AP, pointer(x), BlasInt(stride(x, 1)), convert(T, β), pointer(y), BlasInt(stride(y, 1)))
+    y
+end
+
+"""
+    spmv!(uplo, α, AP, x, β, y)
+
+Update vector `y` as `α*AP*x + β*y` where `AP` is a packed symmetric matrix.
+The storage layout for `AP` is described in the reference BLAS module, level-2 BLAS at
+<http://www.netlib.org/lapack/explore-html/>.
+
+The scalar inputs `α` and `β` shall be non-complex numbers.
```

Comment:
Rewording suggestion:
The scalar inputs `α` and `β` must be either `Float32` or `Float64`.

Reply:
Currently the convert routines called in the short version of the function permit any non-complex number. The long function header is more restrictive and currently undocumented as are all  the other lowest-level-longest-header `blas.jl` functions. So if I would apply this docstring change I would be stating an untrue claim about the short-header function namely that it is actually more restrictive than it is reality. In the long run it would make sense to write separate docstrings for all function headers, but maybe this could be part of a separate commit/pull request?

Reply:
You're right. Then maybe
```julia
The scalar inputs α and β must be real.
```
And you could enforce that actually in the signature by requiring `Real`. My point was mainly that "non-complex" sounds fairly generic.